compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/calc/ConditionalNode.java

```diff
@@ -244,6 +253,17 @@ public static ValueNode canonicalizeConditional(LogicNode condition, ValueNode t
                                 ValueNode and = new AndNode(shift, add.getY());
                                 return new AddNode(add.getX(), and);
                             }
+                        } else if (isAArch64 && trueValue instanceof NegateNode) {
+                            /*
+                             * Detect (x < 0) ? -x : x (and similar variants) and replace with an
+                             * "abs" node. The "absolute" node only applies to AArch64 machines.
+                             */
+                            if (lt.getY().isJavaConstant() && (lt.getY().asJavaConstant().asLong() == 0 || lt.getY().asJavaConstant().asLong() == 1)) {
```

Comment:
The above check for `lt.getY().isDefaultConstant` is equivalent to `lt.getY().isJavaConstant() && lt.getY().asJavaConstant().asLong() == 0` so that check isn't needed here. 

Too I refactored the code so that `lt.getY().isJavaConstant() && lt.getY().asJavaConstant().asLong() == 1` can be triggered